mis/add_binary.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ### 67. Add Binary
+# 
+# https://leetcode.com/problems/add-binary/
+# 
+# Given two binary strings, return their sum (also a binary string).
+# 
+# The input strings are both non-empty and contains only characters 1 or 0.
+# 
+# Example 1:
+# 
+# Input: a = "11", b = "1"
+# Output: "100"
+# Example 2:
+# 
+# Input: a = "1010", b = "1011"
+# Output: "10101"
+
+# In[18]:
+
+
+# swap a, and b so that a is always the shorter length string
+# reversed bit order in each string
+def add_two_bin_strings(a, b):
+    class Solution:
+    def addBinary(self, a: str, b: str) -> str:
+        if len(a) > len(b):
+            b = '0' * (len(a) - len(b)) + b
+        else: 
+            a = '0' * (len(b) - len(a)) + a
+        carry = '0'
+        output = ''
+        for d1, d2 in zip(a[::-1], b[::-1]):
+            if (d1 == '0' and d2 == '1') or (d1 == '1' and d2 == '0'):
```

Comment:
The code below works... However, it is pretty hard to read due to the if-else. Remember whenever you have an `if-else` statement, the reader needs to memorize the condition. If you have several levels of if-else, it becomes challenging. 

https://softwareengineering.stackexchange.com/questions/28238/is-using-else-bad-programming

For this particular case, I suggest to convert each digit to integer. Then, you can add both of them and `carry` module 2 -- just as intuitive as how we add these digits in our head. It'll be much easier to read -- no if-else.

Reply:
You also did not catch the case where the digit is non `0`/`1`.

Reply:
If you did the extension as I asked above (`base_n`) you will very likely need to do `digit_a + digit_b + carry` modulo base.

Reply:
> I remember that it states somewhere in the problem description that I cannot use the function to convert str to int.

I didn't see that anywhere in leetcode. Anyway, my comment was on using `if`/`else`.

Say, if this is not binary but rather base 8. How are you going to do it? Still using if/else?

Reply:
No, I would not use if/else. Neither do I use int('1') and it would not work for hexa-number ('A': 10, 'B':11...). I think that I should use a dict.